tests/registries/test_handler_matching.py

```diff
@@ -30,72 +28,123 @@ def register_fn(registry, resource):
     raise Exception(f"Unsupported registry type: {registry}")
 
 
-@pytest.fixture()
-def cause_no_diff(resource):
-    return Mock(resource=resource, event='some-event', diff=None)
+@pytest.fixture(params=[
+    pytest.param(None, id='without-diff'),
+    pytest.param([], id='with-empty-diff'),
+])
+def cause_no_diff(request, resource):
+    return Mock(resource=resource, event='some-event', diff=request.param)
 
 
-@pytest.fixture()
+@pytest.fixture(params=[
+    pytest.param([('op', ('some-field',), 'old', 'new')], id='with-field-diff'),
+])
 def cause_with_diff(resource):
     diff = [('op', ('some-field',), 'old', 'new')]
     return Mock(resource=resource, event='some-event', diff=diff)
 
 
-def test_catch_all_handlers_found(registry, register_fn, cause_no_diff):
-    register_fn(some_fn, event=None)
-    handlers = registry.get_handlers(cause_no_diff)
+@pytest.fixture(params=[
+    pytest.param(None, id='without-diff'),
+    pytest.param([], id='with-empty-diff'),
+    pytest.param([('op', ('some-field',), 'old', 'new')], id='with-field-diff'),
+])
+def cause_any_diff(resource, request):
+    return Mock(resource=resource, event='some-event', diff=request.param)
+
+#
+# "Catch-all" handlers are those with event == None.
+#
+
+def test_catchall_handlers_without_field_found(cause_any_diff, registry, register_fn):
+    register_fn(some_fn, event=None, field=None)
+    handlers = registry.get_handlers(cause_any_diff)
     assert handlers
 
 
-def test_relevant_event_handlers_found(registry, register_fn, cause_no_diff):
-    register_fn(some_fn, event='some-event')
+def test_catchall_handlers_with_field_found(cause_with_diff, registry, register_fn):
+    register_fn(some_fn, event=None, field='some-field')
+    handlers = registry.get_handlers(cause_with_diff)
+    assert handlers
```

Comment:
Do you assert when this is not empty? Does it mean it catches all the handlers?

Reply:
Yes. It means, that something is there, i.e. non-empty. 

There is only one handler registered in the previous line, so technically it also mean that all handlers are returned (for that field: since both then handler and the cause contain this field — in this specific test).